jerry-core/jerry.c

```diff
@@ -119,526 +124,565 @@ jerry_make_api_available (void)
  *
- * Warning:
- *         acquired pointer should be released with jerry_release_object
- *
- * @return pointer that may be used outside of the engine
+ * @return true  - if the specified value is an array object,
+ *         false - otherwise.
  */
-jerry_object_t *
-jerry_acquire_object (jerry_object_t *object_p) /**< pointer passed to function */
+bool
+jerry_value_is_array (const jerry_value_t value) /**< jerry api value */
 {
   jerry_assert_api_available ();
 
-  ecma_ref_object (object_p);
-
-  return object_p;
-} /* jerry_acquire_object */
+  return (ecma_is_value_object (value)
+          && ecma_get_object_type (ecma_get_object_from_value (value)) == ECMA_OBJECT_TYPE_ARRAY);
+} /* jerry_value_is_array */
 
 /**
- * Release object pointer
+ * Check if the specified value is boolean.
  *
- * See also:
- *          jerry_acquire_object
- *          jerry_call_function
- *          jerry_get_object_field_value
+ * @return true  - if the specified value is boolean,
+ *         false - otherwise.
  */
-void
-jerry_release_object (jerry_object_t *object_p) /**< pointer acquired through jerry_acquire_object */
+bool
+jerry_value_is_boolean (const jerry_value_t value) /**< api value */
 {
-  jerry_assert_api_available ();
-
-  ecma_deref_object (object_p);
-} /* jerry_release_object */
+  return ecma_is_value_boolean (value);
+} /* jerry_value_is_boolean */
 
 /**
- * Acquire specified Jerry API value.
- *
- * Note:
- *      For values of string and object types this acquires the underlying data,
- *      for all other types it is a no-op.
+ * Check if the specified value is a constructor function object value.
  *
- * Warning:
- *         Acquired pointer should be released with jerry_release_value
+ * @return true - if the specified value is a function value that implements [[Construct]],
+ *         false - otherwise.
  */
-jerry_value_t
-jerry_acquire_value (jerry_value_t value) /**< API value */
+bool
+jerry_value_is_constructor (const jerry_value_t value) /**< jerry api value */
 {
   jerry_assert_api_available ();
 
-  if (jerry_value_is_string (value))
-  {
-    jerry_string_t *str_p = jerry_acquire_string (jerry_get_string_value (value));
-    return jerry_create_string_value (str_p);
-  }
-  else if (jerry_value_is_object (value))
-  {
-    jerry_acquire_object (jerry_get_object_value (value));
-  }
-
-  return value;
-} /* jerry_acquire_value */
+  return ecma_is_constructor (value);
+} /* jerry_value_is_constructor */
 
 /**
- * Release specified Jerry API value
+ * Check if the specified value is an error value.
  *
- * Note:
- *      For values of string and object types this releases the underlying data,
- *      for all other types it is a no-op.
+ * @return true  - if the error flag of the specified value is true,
+ *         false - otherwise.
  */
-void
-jerry_release_value (jerry_value_t value) /**< API value */
+bool
+jerry_value_is_error (const jerry_value_t value) /**< api value */
```

Comment:
I would prefer jerry_value_has_error_flag, since strictly speaking, we don't have "error value". Any value can be an error with the error flag.